### PR TITLE
mesa: updating llvm dependency for version 23

### PR DESCRIPTION
**Pull request by renjithravindrankannath**
This change address the issue 42417
Hi @v-dobrev, Please confirm if the dependency is fine.

### CODE REVIEW

**Review by alalazo — APPROVED** [review]
Waiting for @v-dobrev approval before merging

**Review by v-dobrev — APPROVED** [review]

**Pull request merged**
